### PR TITLE
Renamed output to Distance Matrix

### LINKED ISSUE #3082: Widget: Distance matrix -> no output
##### Orange version

3.13

##### Expected behavior
Distance matrix would generate data for scatter plot.

##### Actual behavior
Distance matrix widget does not give output.

##### Steps to reproduce the behavior
Distances -> Distance matrix -> Scatter plot (or Data Table

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Fixes #3082. 

##### Description of changes
Renamed output of Distance Matrix to match other widgets supporting selection.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/unsupervised/owdistancematrix.py [hunk 1]
```diff
@@ -208,7 +208,7 @@ class Inputs:
 
     class Outputs:
         distances = Output("Distances", DistMatrix, dynamic=False)
-        table = Output("Table", Table)
+        table = Output("Selected Data", Table)
```
Comment (astaric) [inline Orange/widgets/unsupervised/owdistancematrix.py]: Consider adding replaces:

https://github.com/biolab/orange3/blob/master/Orange/widgets/utils/signals.py#L110

**Comment by ajdapretnar** [comment]
> If all the tests pass, we need better tests :)

😆 True. I will fix it.

**Comment by ajdapretnar** [comment]
@astaric I think it should work now. Do I need to fix anything else regarding the test workflow? Or does it run automatically?

**Comment by astaric** [comment]
It should run automatically.

If you want to try it out locally, you can run workflow tests with:
```
python -m unittest Orange.widgets.tests.test_workflows -v
```

**Pull request merged**